### PR TITLE
feat: introduce native ExitModule

### LINKED ISSUE #47: Add back navigation in Shipping Zones list
### Description

List of Shipping Zones should have a navigation back button, that would close the React Native embedded view.

### PR DESCRIPTION
**Pull request by wzieba**
Depends on #61 

closes #47 

### Description

I'd like to propose checking how big of an effort would be to introduce iOS native module which would handle back navigation. On Android, as you can see, it was quite easy - without any hacks or non-native feels.

I didn't experiment with iOS though as I don't know right away how to include the Shared `.framework` into WooCommerce Android

@Ecarrion do you mind checking the effort to support back navigation on iOS? Ideally, we would like to include a module _inside_ the Shared library to reduce complexity.

#### Android

https://github.com/woocommerce/WooCommerce-Shared/assets/5845095/fcb1ee4d-31ce-4c17-a9de-d503d3d65903

### CODE REVIEW

File: ShippingZones.tsx [hunk 1]
```diff
@@ -93,6 +95,13 @@ const ShippingZonesList = () => {
 
   useEffect(() => {
     navigation.setOptions({
+      headerLeft: () => <HeaderBackButton
+          style={{marginLeft: Platform.OS === "ios" ? -15 : -5}}
+          labelVisible={Platform.OS === "ios"}
+          label="Settings"
+          onPress={() => {
+            NativeModules.ExitModule.exit();
+          }}/>,
```
Comment (wzieba) [inline ShippingZones.tsx]: I've introduced [`HeaderBackButton`](https://reactnavigation.org/docs/elements/#headerbackbutton) to have back navigation buttons to look native on both Android and iOS.

But, as we use [`Native Stack Navigator`](https://reactnavigation.org/docs/native-stack-navigator), the `HeaderBackButton`, which is default control for the [`Stack Navigator`](https://reactnavigation.org/docs/stack-navigator) is a little off - that's why I needed to set custom margins etc.

After reading more about Native Stack Navigator and Stack Navigator, I think we should consider using former. I've added issue to track that #55 but it's not high priority I believe

**Comment by Ecarrion** [comment]
Hey @wzieba I gave this a try and these are my findings.

- It works well when connected to the metro dev server
- It does not render the back button when running the transpired binary (not in the demo app or in the WCiOS App)
- Those magic margins make me nervous if we plan to run this on iPad or in other orientations.

Metro Server | Transpiled Framework
---- | ----
local-0 | transpiled

I pushed my code if case you want to take a look at it but I think we need to invest more time in it if we want to make it work correctly :-/

**Comment by wzieba** [comment]
> It does not render the back button when running the transpired binary (not in the demo app or in the WCiOS App)

😦 that's surprising, yet another thing to be aware when working with RN - how the code behave if we use metro server vs. if the code is embedded. 

I think about using another navigation stack, maybe those problem will disappear.

**Comment by Ecarrion** [comment]
I may be onto something. I filed an issue and requested help [here](https://github.com/woocommerce/WooCommerce-Shared/issues/61).

**Comment by Ecarrion** [comment]
Good news @wzieba, with #61 it seems like our assets problems on iOS are solved 🎉 

I've updated this branch with it and here are the results.

Metro Server | Demo App
---- | ----
metro-server | demo-app

### WCiOS

https://github.com/woocommerce/WooCommerce-Shared/assets/562080/4d944ea8-cdde-46b8-aff2-6e3b71049fa8

Could you please check that everything still works on Android?

**Comment by wzieba** [comment]
> Good news @wzieba, with https://github.com/woocommerce/WooCommerce-Shared/issues/61 it seems like our assets problems on iOS are solved 🎉

Awesome! From video looks great 👍 

> Could you please check that everything still works on Android?

Yes I confirm - everything still works on Android 👍

**Review by Ecarrion — APPROVED** [review]

**Pull request merged**